app/src/main/java/com/example/travellens/DetailActivity.java

```diff
@@ -0,0 +1,212 @@
+            }
+        });
+        tvUserInDes.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                goToProfile();
+            }
+        });
+
+        // checks whether user has previously liked this post
+        queryIfLiked();
+        queryHowManyLikes();
+        ivLikes.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                likeOrUnlike();
+            }
+        });
+    }
+
+    private void goToProfile() {
+        ProfileFragment profileFragment = new ProfileFragment(thePost.getUser());
+        AppCompatActivity activity = (AppCompatActivity)this;
+        activity.getSupportFragmentManager().beginTransaction().replace(R.id.flContainer, profileFragment).addToBackStack(null).commit();
+    }
+
+    private void likeOrUnlike() {
+        ParseQuery<Likes> query = ParseQuery.getQuery(Likes.class);
+        query.whereEqualTo(Likes.KEY_POST, thePost);
+        query.whereEqualTo(Likes.KEY_USER, ParseUser.getCurrentUser());
+        query.setLimit(1);
+        query.findInBackground(new FindCallback<Likes>() {
+            @Override
+            public void done(List<Likes> results, ParseException e) {
+                if (e == null && !results.isEmpty()) {
+                    for (Likes like : results) {
+                        // deleted from likes
+                        like.deleteInBackground();
+                        // set image to unlike heart
+                        ivLikes.setImageResource(R.drawable.ufi_heart);
+                        // update how many likes the post has
+                        likeCount = likeCount - 1;
+
+                        if (likeCount == 0) {
+                            tvLikes.setText("");
+                        } else if (likeCount == 1) {
+                            tvLikes.setText(getString(R.string.one_like, likeCount));
+                        } else {
+                            tvLikes.setText(getString(R.string.likes, likeCount));
+                        }
+                    }
+                } else if (e != null){
+                    Log.e("Detail Fragment: ", e.toString());
+                } else {
+                    // make a new like object
+                    Likes like = new Likes();
+                    like.setUser(ParseUser.getCurrentUser());
+                    like.setPost(thePost);
+                    // saving it
+                    like.saveInBackground();
+                    // set image to like heart
+                    ivLikes.setImageResource(R.drawable.img_2);
+                    // update how many likes the post has
+                    likeCount = likeCount + 1;
+                    if (likeCount == 0) {
+                        tvLikes.setText("");
+                    } else if (likeCount == 1) {
+                        tvLikes.setText(getString(R.string.one_like, likeCount));
```

Comment:
You should take a look at string plurals. That will have you not have to check for the likeCount being 1 or more than 1 in the code.